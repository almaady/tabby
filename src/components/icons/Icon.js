import React from "react";
import PropTypes from "prop-types";

/**
 * @class Icon - Is the component to use icons from our font-icon
 */
class Icon extends React.PureComponent {
  /**
   * @property {string} icon - The function to close the modal
   * @property {string} size - to define icon's size
   * @property {string} color - To define icon's color
   */
  static propTypes = {
    icon: PropTypes.string,
    size: PropTypes.string,
    color: PropTypes.string
  };

  render() {
    const { icon, size, color } = this.props;
    return <i className={`icon-${icon} color--${color} size--${size} icon--height`} />;
  }
}

Icon.defaultProps = {
  size: "XXL",
  color: "gray-l1",
  icon:'cross'
};

export default Icon;
