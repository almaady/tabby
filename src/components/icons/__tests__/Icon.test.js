import React from "react";
import { shallow } from "enzyme";

import Icon from "../Icon";

describe("Icon", () => {
  let props;
  beforeEach(() => {
    props = {
      icon: ""
    };
  });

  it("should render correctly with default state", () => {
    const component = shallow(<Icon {...props} />);
    expect(component).toMatchSnapshot();
  });
  it("should add color--red class when color prop is red",() =>{
    const component = shallow(<Icon {...props} color="red"/>);
    expect(component.hasClass("color--red")).toEqual(true);
  })
});
