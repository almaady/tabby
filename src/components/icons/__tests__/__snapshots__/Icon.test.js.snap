// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly with default state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Icon
    color="gray-l1"
    icon=""
    size="XXL"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "icon- color--gray-l1 size--XXL icon--height",
    },
    "ref": null,
    "rendered": null,
    "type": "i",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "icon- color--gray-l1 size--XXL icon--height",
      },
      "ref": null,
      "rendered": null,
      "type": "i",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
